Lib/unittest/test/testmock/testpatch.py

```diff
@@ -105,6 +105,15 @@ def test():
         self.assertEqual(Something.attribute, sentinel.Original,
                          "patch not restored")
 
+    def test_patchobject_with_string_as_target(self):
+        class Something(object):
+            def do_something(self):
+                pass
+
+        def test():
+            with patch.object('Something', 'do_something'):
+                pass
+        self.assertRaises(TypeError, test)
```

Comment:
Can probably simplify this test by replacing the `test` function with

```python
self.assertRaises(TypeError, patch.object, 'Something', 'do_something')
```

also, since this is just checking for an error condition, `class Something` can be be removed too.